db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/subquery/GroupByUtil.java

```diff
@@ -62,23 +67,94 @@
 
         /*
          * Nominal case: subquery is correlated, has one or more correlated predicates.  We will group by the subquery
          * columns that are compared to the outer query columns.
          */
+        ColumnReference outerColumnReference = null;
         if (!correlatedSubqueryPreds.isEmpty()) {
             int subqueryNestingLevel = subquerySelectNode.getNestingLevel();
 
             for (BinaryRelationalOperatorNode bro : correlatedSubqueryPreds) {
                 if (PredicateUtils.isLeftColRef(bro, subqueryNestingLevel)) {
                     addGroupByNodes(subquerySelectNode, bro.getLeftOperand());
+                    outerColumnReference = (ColumnReference)bro.getRightOperand();
                 } else if (PredicateUtils.isRightColRef(bro, subqueryNestingLevel)) {
                     addGroupByNodes(subquerySelectNode, bro.getRightOperand());
-                } else {
+                    outerColumnReference = (ColumnReference)bro.getLeftOperand();
+                }
+            }
+        }
+        else if (!correlatedSubqueryInequalityPreds.isEmpty()) {
+            for (BinaryRelationalOperatorNode bro : correlatedSubqueryInequalityPreds) {
+                CorrelatedColRefCollectingVisitor correlatedColRefCollectingVisitor
+                    = new CorrelatedColRefCollectingVisitor<>(1, outerNestingLevel);
+                bro.accept(correlatedColRefCollectingVisitor);
+                if (correlatedColRefCollectingVisitor.getCollected().isEmpty())
                     throw new IllegalArgumentException("Did not find correlated column ref on either side of BRON");
+                else {
+                        ColumnReference leftColRef, rightColRef, correlatedColumnReference;
+                        correlatedColumnReference =
+                            (ColumnReference)correlatedColRefCollectingVisitor.getCollected().get(0);
+
+                        FromTable fromTable;
+                        String tableName = correlatedColumnReference.getTableName();
+                        String schemaName = correlatedColumnReference.getSchemaName();
+                        if (tableName == null || schemaName == null) {
+                            fromTable = outerSelectNode.getFromList().getFromTableByTableNumber(correlatedColumnReference.getTableNumber());
+                            correlatedColumnReference.setTableNameNode(fromTable.getTableName());
+                        }
+                        else
+                            fromTable = outerSelectNode.getFromList().
+                                             getFromTableByName(correlatedColumnReference.getTableName(),
+                                                                correlatedColumnReference.getSchemaName(), true);
+                        leftColRef = (ColumnReference)correlatedColumnReference.getClone();
+                        rightColRef = (ColumnReference)correlatedColumnReference.getClone();
```

Comment:
Change the names to subqueryColRef and outerColumnReference and added this comment:

                // subqueryColRef is the version of the correlatedColumnReference pushed into the join
                //                in the new generated outer_table.col = inner_table.col predicate.
                // outerColumnReference is the version of the correlatedColumnReference
                //                      in the new generated outer_table.col = inner_table.col predicate,
                //                      but cloned so that it has memory separate from the original
                //                      correlated inequality predicate.